metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -1125,183 +1129,191 @@ def create_test_entity_mcps(
-                        entityType="assertion",
-                        entityUrn=node_datahub_urn,
+                    self.report.report_workunit(wu)
+                    yield wu
+
+                if self.config.delete_tests_as_datasets:
+                    mce_platform = (
+                        self.config.target_platform
+                        if self.config.disable_dbt_node_creation
+                        else DBT_PLATFORM
+                    )
+                    node_datahub_urn = get_urn_from_dbtNode(
+                        node.database,
+                        node.schema,
+                        node.alias or node.name,  # previous code used the alias
+                        mce_platform,
+                        self.config.env,
+                        self.config.platform_instance
+                        if mce_platform == DBT_PLATFORM
+                        else None,
+                    )
+                    soft_delete_mcp = MetadataChangeProposalWrapper(
+                        entityType="dataset",
                         changeType=ChangeTypeClass.UPSERT,
-                        aspectName="assertionInfo",
-                        aspect=assertion_info,
-                    ),
-                )
-                self.report.report_workunit(wu)
-                yield wu
-
-            if self.config.delete_tests_as_datasets:
-                mce_platform = (
-                    self.config.target_platform
-                    if self.config.disable_dbt_node_creation
-                    else DBT_PLATFORM
-                )
-                node_datahub_urn = get_urn_from_dbtNode(
-                    node.database,
-                    node.schema,
-                    node.alias or node.name,  # previous code used the alias
-                    mce_platform,
-                    self.config.env,
-                    self.config.platform_instance
-                    if mce_platform == DBT_PLATFORM
-                    else None,
-                )
-                soft_delete_mcp = MetadataChangeProposalWrapper(
-                    entityType="dataset",
-                    changeType=ChangeTypeClass.UPSERT,
-                    entityUrn=node_datahub_urn,
-                    aspectName="status",
-                    aspect=StatusClass(removed=True),
-                )
-                soft_delete_wu = MetadataWorkUnit(
-                    id=f"{node_datahub_urn}-status", mcp=soft_delete_mcp
+                        entityUrn=node_datahub_urn,
+                        aspectName="status",
+                        aspect=StatusClass(removed=True),
+                    )
+                    soft_delete_wu = MetadataWorkUnit(
+                        id=f"{node_datahub_urn}-status", mcp=soft_delete_mcp
+                    )
+                    self.report.report_workunit(soft_delete_wu)
+                    yield soft_delete_wu
+
+            if self.config.test_results_path:
```

Comment:
seems repetitive to have this code fragment twice

Reply:
I thought it's a good idea to delegate ingestion of assertion to a different function 
Because right now in `create_test_entity_mcps` we have the creation of an assertion object based on manifest and source files. But when we want only to ingest the result of assertions, we don't need to create an assertion object once again but only append the result.